### ISSUE TITLE
Aspera downloading with ascli using the Proxy configuration

### ISSUE THREAD

**Issue by xavierv79** [IBM/aspera-cli#144]
Hi Team,

We are using "ascli " to download files from aspera with https gateway. But additionally we need the  proxy configuration of network to be passed with below.

```bash
ascli shares files download --to-folder=. share/file --transfer=httpgw --transferinfo=@json:'{"url":"https://example.com/aspera/http-gwy"}’
```

Please provide the correct syntax to pass the  proxy IP, port , username and pass with above command.

**Comment by laurent-martin** [comment]
This is the correct syntax, but I notice that the `-` in. `--transfer-info` in your example is wrong. Makefile it is ASCII hyphen character, not unicode.

Also, are you executing on a shell like bash , or on window ?

Also, not that if the user is always using a proxy, he can define this in the configuration file.

**Comment by xavierv79** [comment]
Hi Laurent,
Thank you We are using in Linux bash. 
Need your help additionally for the Proxy. syntax to pass the proxy IP, port , username and pass with above command

**Comment by laurent-martin** [comment]
Check documentation here:
https://github.com/IBM/aspera-cli?tab=readme-ov-file#fasp-proxy-forward-for-transfers

syntax is:

```json
{"proxy":"dnat://username:password@proxy.example.org:9091"}
```

**Comment by laurent-martin** [comment]
So, do you want to use both HTTP Gateway and proxy ????

**Comment by xavierv79** [comment]
Hi Laurent,
We need to use both HTTP Gateway and  proxy, since Internet and allowed ports will be through proxy.

**Comment by laurent-martin** [comment]
I see,
Well, typically, if an http gateway is used, the http gateway has direct internet connection.
if the use cannot reach the http gateway, then he can use an http proxy to reach the http gateway.
Now, if the http gateway does not have internet access, and needs itself to use a forward fasp proxy .. Mmm...

**Comment by laurent-martin** [comment]
Here typical paths:

<img width="1188" alt="image" src="https://github.com/IBM/aspera-cli/assets/11157566/8f6526f5-85de-45d1-9fbd-17db83a1aeae">

**Comment by laurent-martin** [comment]
Are you asking for the purple path ?

<img width="865" alt="image" src="https://github.com/IBM/aspera-cli/assets/11157566/1880effb-9891-4a3e-b369-8676d6fafd98">

**Comment by xavierv79** [comment]
Hi Laurent,

Here Aspera Server is ours and the end user need to use their Proxy . End user need to use their proxy with our http gateway during the https transfer through ascli.

**Comment by laurent-martin** [comment]
If the end user has an **Aspera Forward Proxy**, it does not make sense because it means they will use FASP, and so do not need the HTTP Gateway.

If the remote user needs to use the HTTP Gateway, mays they have an HTTP proxy (not aspera proxy) , can you confirm you talk about HTTP proxy on client side ?
If so, the doc is:

https://github.com/IBM/aspera-cli?tab=readme-ov-file#proxy-for-rest-and-http-gateway

Basically, `ascli` uses Ruby's HTTP library, and proxy rules apply.

**Comment by xavierv79** [comment]
Hi Laurent,

Yes it's like  purple path but instead of HSTS they want use aspera gateway, since aspera connect port not allowed form their end.

**Comment by laurent-martin** [comment]
Ok, yes, the purple path shows: ascli -> HTTP forward proxy -> HTTP GW -> HSTS

on bash, the env var can also be provided on the command line like this:

```bash
http_proxy=http://username:password@host:port ascli shares files download --to-folder=. share/file --transfer=httpgw --transfer-info=@json:'{"url":"https://example.com/aspera/http-gwy"}’
```

**Comment by laurent-martin** [comment]
Note: next version (not yet released) will have option `--http-proxy=` which behaves like env var `http_proxy`.

**Comment by xavierv79** [comment]
Hi Laurent,
Thank for the Update. Please let me know when the version would be released approximately.

**Comment by laurent-martin** [comment]
@xavierv79 , did you test version 4.18.1 ?
